core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/cluster/datalake/upgrade/ClusterUpgradeService.java

```diff
@@ -0,0 +1,69 @@
+package com.sequenceiq.cloudbreak.core.flow2.cluster.datalake.upgrade;
+
+import static com.sequenceiq.cloudbreak.event.ResourceEvent.CLUSTER_MANAGER_UPGRADE;
+import static com.sequenceiq.cloudbreak.event.ResourceEvent.CLUSTER_MANAGER_UPGRADE_FAILED;
+import static com.sequenceiq.cloudbreak.event.ResourceEvent.CLUSTER_MANAGER_UPGRADE_FINISHED;
+import static com.sequenceiq.cloudbreak.event.ResourceEvent.CLUSTER_UPGRADE;
+import static com.sequenceiq.cloudbreak.event.ResourceEvent.CLUSTER_UPGRADE_FAILED;
+import static com.sequenceiq.cloudbreak.event.ResourceEvent.CLUSTER_UPGRADE_FINISHED;
+import static com.sequenceiq.cloudbreak.event.ResourceEvent.DATALAKE_UPGRADE;
+
+import javax.inject.Inject;
+
+import org.springframework.stereotype.Service;
+
+import com.sequenceiq.cloudbreak.api.endpoint.v4.common.DetailedStackStatus;
+import com.sequenceiq.cloudbreak.api.endpoint.v4.common.Status;
+import com.sequenceiq.cloudbreak.cloud.model.catalog.Image;
+import com.sequenceiq.cloudbreak.core.flow2.stack.CloudbreakFlowMessageService;
+import com.sequenceiq.cloudbreak.service.StackUpdater;
+import com.sequenceiq.cloudbreak.service.cluster.ClusterService;
+import com.sequenceiq.cloudbreak.service.image.StatedImage;
+
+@Service
+public class ClusterUpgradeService {
```

Comment:
This is a simple class but if an issue comes up later because some developer did something wrong here then everybody will scream the developer's name who developed the cluster upgrade flow and unfortunately the support guy, or the e2e guy, or you should spend hours to find the cause and then that lazy guy, who broke the  functionality and didn't test his work, so I would write a unit test to make his job harder. This is true for other classes as well.